app/Models/Prefix.php

```diff
@@ -27,4 +28,34 @@ public function detachAllFormats()
   {
     $this->formats()->detach();
   }
+
+  public static function findPrefixLabel($formatId, $collectionId)
+  {
+    $collectionTypeIdQuery = DB::table('collections')->select('collection_type_id')
+                                                     ->where('id', '=', $collectionId)
+                                                     ->get()
+                                                     ->first();
+
+    $collectionTypeId = $collectionTypeIdQuery === null ? null : $collectionTypeIdQuery->collection_type_id;
+
+    # find the prefix attached to the specified format
+    # that has the same collection type ID as the specified collection
+    $labelQuery = DB::table('prefixes')->select('label')
+                                       ->join('format_prefix', 'prefixes.id', '=', 'format_prefix.prefix_id')
+                                       ->where([
+                                         ['format_prefix.format_id', '=', $formatId],
+                                         ['prefixes.collection_type_id', '=', $collectionTypeId]
+                                       ])
+                                       ->get()
+                                       ->first();
+
+    $label = $labelQuery === null ? null : $labelQuery->label;
```

Comment:
Stop checking for null or throw exception for null once you're ready to remove the legacy code below.